packages/tradershub/src/features/cfd/screens/MT5TradeScreen/MT5TradeLink/MT5TradeLink.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { FC } from 'react';
+import { useActiveTradingAccount, useCtraderServiceToken } from '@deriv/api';
+import { Button, Text } from '@deriv/quill-design';
+import { getPlatformFromUrl } from '../../../../../helpers/urls';
+import { THooks, TPlatforms } from '../../../../../types';
+import { AppToContentMapper, PlatformDetails, PlatformToLabelIconMapper, PlatformUrls } from '../../../constants';
+
+type TMT5TradeLinkProps = {
+    app?: keyof typeof AppToContentMapper;
+    platform?: TPlatforms.All;
+    webtraderUrl?: THooks.MT5AccountsList['webtrader_url'];
+};
+
+const MT5TradeLink: FC<TMT5TradeLinkProps> = ({ app = 'linux', platform, webtraderUrl = '' }) => {
+    const content = AppToContentMapper[app];
+    const { data: ctraderToken } = useCtraderServiceToken();
+
+    const { data: activeAccount } = useActiveTradingAccount();
+
+    const isDemo = activeAccount?.is_virtual;
+
+    const mt5Platform = PlatformDetails.mt5.platform;
+    const dxtradePlatform = PlatformDetails.dxtrade.platform;
+    const ctraderPlatform = PlatformDetails.ctrader.platform;
+
+    const onClickWebTerminal = () => {
+        const { isStaging, isTestLink } = getPlatformFromUrl();
+        let url;
+        switch (platform) {
+            case dxtradePlatform:
+                url = isDemo ? PlatformUrls.dxtrade.demo : PlatformUrls.dxtrade.live;
+                break;
+            case ctraderPlatform:
+                url = isTestLink || isStaging ? PlatformUrls.ctrader.staging : PlatformUrls.ctrader.live;
+                if (ctraderToken) url += `?token=${ctraderToken}`;
+                break;
+            default:
+                url = '';
+        }
```

Comment:
I feel this can be a nested object with a key

Reply:
that is assuming both their urls are **demo** and **live**, it is possible do implement something like `{platform}.{account_type}`

however in the scenario here, dxtrade has `demo` and `live` url while ctrader has `staging` and `live` url

Reply:
Staging URL for Ctrader is UAT Environment which is accessible on basis of Production or Non-Production (like Staging, Testlink, Localhost)
The url for demo and Live is same.

Reply:
<img width="310" alt="image" src="https://github.com/binary-com/deriv-app/assets/108507236/3dfafa60-2755-4bf3-b97a-e9baac52140b">
referring to the styles from wallets